llvm/lib/Target/X86/X86InstrInfo.h

```diff
@@ -574,8 +574,8 @@ class X86InstrInfo final : public X86GenInstrInfo {
                      outliner::Candidate &C) const override;
 
   void buildClearRegister(Register Reg, MachineBasicBlock &MBB,
-                          MachineBasicBlock::iterator Iter,
-                          DebugLoc &DL) const override;
+                          MachineBasicBlock::iterator Iter, DebugLoc &DL,
+                          bool NoSideEffects = false) const override;
```

Comment:
who calls buildClearRegister with this new default param set to true?

also, I fear double negatives here. Would it be clearer to call this `SideEffects` and default to `true`?

Reply:
I'm going to be using this for clearing registers using stack protectors. I don't think it's clearer to use `SideEffects`. I know it's a double negative, but it just feels more natural to me to say "I want the instruction to have no side effects."

Reply:
Perhaps it should be `AvoidSideEffects`? Then it wouldn't not haven't had that double negative...

Reply:
Yeah, that SGTM